packages/thirdweb/src/extensions/prebuilts/deploy-vote.ts

```diff
@@ -0,0 +1,186 @@
+import type { Chain } from "src/chains/types.js";
+import type { ThirdwebClient } from "../../client/client.js";
+import { type ThirdwebContract, getContract } from "../../contract/contract.js";
+import { deployViaAutoFactory } from "../../contract/deployment/deploy-via-autofactory.js";
+import { getOrDeployInfraForPublishedContract } from "../../contract/deployment/utils/bootstrap.js";
+import type { FileOrBufferOrString } from "../../storage/upload/types.js";
+import type { Prettify } from "../../utils/type-utils.js";
+import type { ClientAndChainAndAccount } from "../../utils/types.js";
+import { decimals } from "../erc20/read/decimals.js";
+
+/**
+ * @extension PREBUILT
+ */
+export type VoteContractParams = {
+  name: string;
+  /**
+   * The contract address for the ERC20 that will be used as voting power
+   */
+  token: string;
```

Comment:
be super explicit on naming - this is your public API, once its out there its very hard to change.

if this is an address - call it `tokenAddress`. There should be no doubt about what this is by reading the variable name